src/webgpu/api/validation/vertex_access.spec.ts

```diff
@@ -1,91 +1,483 @@
+In this test we test that any missing buffer for a used slot will cause validation errors when drawing.
+- All (non/indexed, in/direct) draw commands
+    - A needed vertex buffer is not bound
+        - Was bound in another render pass but not the current one
+        - x= all vertex formats
+- Indexed draw commands,
+    - No index buffer is bound
+`
+  )
+  .unimplemented();
+
+g.test(`unused_buffer_bound`)
+  .desc(
+    `
+In this test we test that a small buffer bound to unused buffer slot won't cause validation error.
+- All draw commands,
+  - An unused {index , vertex} buffer with uselessly small range is bound (immediately before draw
+    call)
+`
+  )
+  .unimplemented();
+
+g.test(`last_buffer_setting_take_account`)
+  .desc(
+    `
+In this test we test that only the last setting for a buffer slot take account.
+- All (non/indexed, in/direct) draw commands
+  - setPl, setVB, setIB, draw, {setPl,setVB,setIB,nothing (control)}, then a larger draw that
+    wouldn't have been valid before that
+`
+  )
+  .unimplemented();
+
+g.test(`index_buffer_OOB`)
+  .desc(
+    `
+In this test we test that index buffer OOB is catched as validation error in drawIndex.
+drawIndexedIndirect didn't has such validation yet.
+- Indexed draw commands,
+    - Draw call needs to read {=, >} the bound index buffer range, with GPUBuffer that is {large
+      enough, exactly the size of bound range}
+        - range is too small and GPUBuffer is large enough
+        - range and GPUBuffer are exact size
+        - x= all index formats
+`
+  )
+  .unimplemented();
+
+g.test(`vertex_buffer_OOB`)
+  .desc(
+    `
+In this test we test that vertex buffer OOB is catched as validation error in draw call. Specifically,
+only vertex step mode buffer OOB in draw and instance step mode buffer OOB in draw and drawIndexed
+are CPU-validated. Other cases are currently handled by robust access.
+- Test that:
+    - Draw call needs to read {=, >} any bound vertex buffer range, with GPUBuffer that is {large
+      enough, exactly the size of bound range}
+        - x= all vertex formats
+        - x= weird offset values
+        - x= weird arrayStride values
+        - x= {render pass, render bundle}
+- For vertex step mode vertex buffer,
+    - Test with draw:
+        - vertexCount largeish
+        - firstVertex {=, >} 0
+    - drawIndexed, draIndirect and drawIndexedIndirect are dealt by robust access
+- For instance step mode vertex buffer,
+    - Test with draw and drawIndexed:
+        - instanceCount largeish
+        - firstInstance {=, >} 0
+    - draIndirect and drawIndexedIndirect are dealt by robust access
+`
+  )
+  .unimplemented();
+
+g.test(`largeish_buffer`)
+  .desc(
+    `
+In this test we test that a very large range of buffer is bound to different slot, and no validation
+error occurs.
+- A bound vertex buffer range is significantly larger than necessary
+- A bound index buffer range is significantly larger than necessary
+`
+  )
+  .unimplemented();
```

Comment:
Technically it doesn't matter if this is a validation or operation test, but I think it would make more sense to categorize it as an operation test, e.g. in `api/operation/rendering/draw.spec.ts`.